src/accountsdb/buffer_pool.zig

```diff
@@ -53,208 +104,85 @@ const FileIdFileOffset = packed struct(u64) {
 /// A frame dies when its index is evicted from HierarchicalFifo (inside of
 /// evictUnusedFrame).
 pub const BufferPool = struct {
-    pub const FrameMap = std.AutoHashMapUnmanaged(FileIdFileOffset, FrameIndex);
-
-    /// indices of all free frames
-    /// free frames have a refcount of 0 *and* have been evicted
-    free_list: AtomicStack(FrameIndex),
-
-    /// uniquely identifies a frame
-    /// for finding your wanted index
-    /// TODO: a concurrent hashmap would be more appropriate
-    frame_map_rw: sig.sync.RwMux(FrameMap),
-
     frames: []Frame,
-    frames_metadata: FramesMetadata,
+    manager: Manager,
 
-    /// used for eviction to free less popular (rc=0) frames first
-    eviction_lfu: HierarchicalFIFO,
-
-    /// NOTE: we might want this to be a threadlocal for best performance? I don't think this field is threadsafe
-    io_uring: if (use_io_uring) std.os.linux.IoUring else void,
+    pub const ReadBlockingError = FrameManager.GetError || std.posix.PReadError;
+    pub const ReadIoUringError = FrameManager.GetError || IoUringError;
+    pub const ReadError = if (use_io_uring) ReadIoUringError else ReadBlockingError;
+    pub const Manager = FrameManager;
```

Comment:
should probs reduce rexports if we dont neeeeed them

Reply:
removed Manager re-export https://github.com/Syndica/sig/commit/f40689214b7f8a0945d24c5a2847c29af6db0203. If we want we could only export ReadError?

Reply:
im okay either way